contrib/containers/ci/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:focal
+FROM ubuntu:jammy
```

Comment:
`focal` is still supported, it's 5 years, until April 2025.

can we still keep `focal` for C++20? is gcc-10 available for it?

Reply:
Bitcoin lets to use for different testing targets to use different ubuntu version, and even not ubuntu, but something like CentOS (in the past at least).
We have an option to use the one docker image for all CI tests. So, if we test with newer version ubuntu - the older one is not tested (and we won't know if build is broken). For newer version of ubuntu (the latest one) usually I run builds on my localhost :) 

@UdjinM6 any though if we can use different docker images for different CI tests? I'd like to see the oldest supported (20.04 probably now) and the newest release (23.10 which will be superseeded by 24.04 soon).

could we get `g++-mingw-w64-x86-64` version 10 from the third party repo maybe?